src/cli/service/serve.rs

```diff
@@ -0,0 +1,362 @@
+        let tls_cfg: Option<TlsAcceptorData> = match self.tls_server_config {
+            Some(cfg) => Some(cfg.try_into()?),
+            None => None,
+        };
+
+        let tcp_service_builder = (
+            ConsumeErrLayer::trace(tracing::Level::DEBUG),
+            (self.concurrent_limit > 0)
+                .then(|| LimitLayer::new(ConcurrentPolicy::max(self.concurrent_limit))),
+            (!self.timeout.is_zero()).then(|| TimeoutLayer::new(self.timeout)),
+            tcp_forwarded_layer,
+            BodyLimitLayer::request_only(self.body_limit),
+            #[cfg(any(feature = "rustls", feature = "boring"))]
+            tls_cfg.map(|cfg| {
+                #[cfg(feature = "boring")]
+                return TlsAcceptorLayer::new(cfg).with_store_client_hello(true);
+                #[cfg(all(feature = "rustls", not(feature = "boring")))]
+                TlsAcceptorLayer::new(cfg).with_store_client_hello(true)
+            }),
+        );
+
+        let http_transport_service = match self.http_version {
+            Some(Version::HTTP_2) => Either3::A(HttpServer::h2(executor).service(http_service)),
+            Some(Version::HTTP_11 | Version::HTTP_10 | Version::HTTP_09) => {
+                Either3::B(HttpServer::http1().service(http_service))
+            }
+            Some(_) => {
+                return Err(OpaqueError::from_display("unsupported http version").into_boxed());
+            }
+            None => Either3::C(HttpServer::auto(executor).service(http_service)),
+        };
+
+        Ok(tcp_service_builder.into_layer(http_transport_service))
+    }
+
+    /// build an http service ready to serve files
+    pub fn build_http(
+        &self,
+    ) -> Result<
+        impl Service<(), Request, Response: IntoResponse, Error = Infallible> + use<H>,
+        BoxError,
+    > {
+        let http_forwarded_layer = match &self.forward {
+            None | Some(ForwardKind::HaProxy) => None,
+            Some(ForwardKind::Forwarded) => Some(Either7::A(GetForwardedHeadersLayer::forwarded())),
+            Some(ForwardKind::XForwardedFor) => {
+                Some(Either7::B(GetForwardedHeadersLayer::x_forwarded_for()))
+            }
+            Some(ForwardKind::XClientIp) => {
+                Some(Either7::C(GetForwardedHeadersLayer::<XClientIp>::new()))
+            }
+            Some(ForwardKind::ClientIp) => {
+                Some(Either7::D(GetForwardedHeadersLayer::<ClientIp>::new()))
+            }
+            Some(ForwardKind::XRealIp) => {
+                Some(Either7::E(GetForwardedHeadersLayer::<XRealIp>::new()))
+            }
+            Some(ForwardKind::CFConnectingIp) => {
+                Some(Either7::F(GetForwardedHeadersLayer::<CFConnectingIp>::new()))
+            }
+            Some(ForwardKind::TrueClientIp) => {
+                Some(Either7::G(GetForwardedHeadersLayer::<TrueClientIp>::new()))
+            }
+        };
+
+        let serve_service = match self.content_path {
```

Comment:
```suggestion
        let serve_service = match &self.content_path {
```

this way you do not need to use `ref path` everywhere down :)